packages/tre/src/plugins/core/index.ts

```diff
@@ -99,8 +108,45 @@ export const SCRIPT_CUSTOM_SERVICE = `addEventListener("fetch", (event) => {
   event.respondWith(${BINDING_SERVICE_LOOPBACK}.fetch(request));
 })`;
 
+const now = new Date();
+const CURRENT_COMPATIBILITY_DATE = [
+  now.getFullYear(),
+  (now.getMonth() + 1).toString().padStart(2, "0"),
+  now.getDate().toString().padStart(2, "0"),
+].join("-");
+
 const FALLBACK_COMPATIBILITY_DATE = "2000-01-01";
 
+function validateCompatibilityDate(compatibilityDate: string) {
+  const workerdVersion = getWorkerdVersion();
+  if (numericCompare(compatibilityDate, CURRENT_COMPATIBILITY_DATE) > 0) {
+    // If this compatibility date is in the future, throw
+    throw new MiniflareCoreError(
+      "ERR_FUTURE_COMPATIBILITY_DATE",
+      `Compatibility date "${compatibilityDate}" is in the future and unsupported`
+    );
+  } else if (numericCompare(compatibilityDate, workerdVersion) > 0) {
+    // If this compatibility date is greater than the maximum supported
+    // compatibility date of the runtime, but not in the future, warn,
+    // and use the maximum supported date instead
+    console.warn(
+      yellow(
+        [
+          "The newest compatibility date supported by the installed version of the Cloudflare Workers Runtime is ",
+          bold(`"${workerdVersion}"`),
```

Comment:
What about checking @GregBrimble's new compatibility date API here, to see if they're running a version of workerd that is out of date, and suggest updating wrangler if so?